app/models/user.rb

```diff
@@ -0,0 +1,29 @@
+class User
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :name, String, required: true
+  property :username, String, required: true, unique: true, message: 'Username is already taken'
+  property :email, String, unique: true, required: true, format: :email_address
+  property :password_digest, Text, required: true
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password, message: "Please enter matching passwords"
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(email, password)
+    user = User.first(email: email)
+    if user && BCrypt::Password.new(user.password_digest) == password
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`. (https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier)